client/my-sites/sharing/buttons/options.jsx

```diff
@@ -95,41 +107,39 @@ module.exports = React.createClass( {
 				label: this.getPostTypeLabel( postType )
 			};
 		}, this );
-	},
+	}
 
-	isTwitterButtonEnabled: function() {
+	isTwitterButtonEnabled() {
 		return some( this.props.buttons, { ID: 'twitter', enabled: true } );
-	},
-
-	getTwitterViaOptionElement: function() {
-		var option;
+	}
 
-		if ( ! this.isTwitterButtonEnabled() || ( this.props.site.jetpack && this.props.site.versionCompare( '3.4-dev', '<' ) ) ) {
+	getTwitterViaOptionElement() {
+		if ( ! this.isTwitterButtonEnabled() || ! this.props.isTwitterButtonAllowed ) {
```

Comment:
Since we have `this.props` five places in this method, do you think we should deconstruct the attributes we need into `const`s?